### PR TITLE
Publication sync check fixes

### PR DESCRIPTION
**Pull request by boffbowsh**
This firstly fixes the publication check to fetch the publication type correctly. Previously the checks would crash completely as it was checking a draft publication, and would try to get the type of a nil Edition.

Also adds a check for the withdrawn_at field. This will fail on most withdrawn publications as currently Whitehall is unable to send this field to the Publishing API.

### CODE REVIEW

File: test/unit/sync_checker/checks/unpublished_check_test.rb [hunk 1]
```diff
@@ -90,6 +103,36 @@ def test_returns_an_error_if_the_document_is_withdrawn_and_has_no_notice
...
+          updated_at: Time.mktime(2016, 8, 10, 1, 2, 3, 456)
+        ),
+        pre_publication_edition: nil
+      )
+      document.published_edition.stubs(unpublishing: stub(
+        unpublishing_reason_id: 5,
+        explanation: "Some explainings",
+        edition: document.published_edition
+      )
+                                      )
```
Comment (gpeng) [inline test/unit/sync_checker/checks/unpublished_check_test.rb]: Some funky bracket business going on here...

**Comment by gpeng** [comment]
There are some tests failing with time zone stuff. In the other formats we've been doing lately we've started sending them in UTC which we don't seem to have been doing historically.

**Comment by boffbowsh** [comment]
Fix0red

**Comment by boffbowsh** [comment]
Fixed.

**Pull request merged**